source/common/http/conn_pool_base.h

```diff
@@ -128,12 +128,13 @@ class ActiveClient : public Envoy::ConnectionPool::ActiveClient {
     real_host_description_ = data.host_description_;
     codec_client_ = parent.createCodecClient(data);
     codec_client_->addConnectionCallbacks(*this);
+    InstantiatedClusterStats& stats = *parent_.host()->cluster().stats();
```

Comment:
note that we can do the lazy instantiation once here and cover all the stats below.